### PR TITLE
Add module tags to docs

### PR DESCRIPTION
**Pull request by bashtage**
@TomAugspurger @josef-pkt Here is what I think are the final set of doc fixes. I still can't get the images to generate in the right place. Very frustrating. I think it has something to do with bui…  View pull request

### CODE REVIEW

File: docs/source/glm.rst [hunk 1]
```diff
@@ -180,11 +184,8 @@ available link functions can be obtained by
    NegativeBinomial
    Power
    cauchy
-   cloglog
    identity
    inverse_power
    inverse_squared
-   log
-   logit
```
Comment (josef-pkt) [inline docs/source/glm.rst]: I think those are needed, but they only work on case sensitive operating systems, i.e. not on Windows

Comment (bashtage) [inline docs/source/glm.rst]: ic

Comment (bashtage) [inline docs/source/glm.rst]: Are these different from their upper case equivs?

Comment (josef-pkt) [inline docs/source/glm.rst]: part of this was ancient compatibility and refactoring

log and logit are subclasses of the capitalized that don't change anything, i.e. identical and redundant classes
some other lower case like inverse_power are subclasses that fix a parameter (e.g. power)

There is a partial refactoring in master but not in 0.8, for how links are handled, but no name deprecation yet.
AFAIR, most of our examples use the lowercase `log` but in future we might want to switch to the uppercase class names.

**Comment by bashtage** [comment]
I'll build on Linux and make sure these go away.  I understand why this is a problem on Windows.  I think you mentioned this in the past -- probably best to try to prevent this in the future.

File: docs/source/glm.rst (hunk 1 above)
Comment (josef-pkt) [inline docs/source/glm.rst]: (aside about the ancient part: lower case were instances at the beginning, but that messed up the code with mutable keyword arguments.)

**Comment by josef-pkt** [comment]
about the lowercase, uppercase conflict: In the past I just ignored those warnings during sphinx build on windows. IIRC there were a few more besides the `links`
We have avoided lowercase-uppercase name conflicts since after the beginning. but the code where the old cases are was never seriously refactored.

**Comment by bashtage** [comment]
I made one myself `pca.pca` and `pca.PCA`

**Comment by bashtage** [comment]
This is probably OK now.

**Comment by josef-pkt** [comment]
I don't know much about other changes here.

merging, this is the last on the 0.8 schedule then

**Pull request merged**